core/tests/protractor_utils/editor.js

```diff
@@ -227,6 +228,152 @@ var expectCannotDeleteInteraction = function() {
     '.protractor-test-delete-interaction')).isPresent()).toBeFalsy();
 };
 
+// GADGETS
+
+// Additional arguments may be sent to this function, and they will be
+// passed on to the relevant gadget editor.
+var addGadget = function(panelName, gadgetType, gadgetName) {
+
+  // Bring up the gadget insertion modal for the specified panel.
+  element(
+    by.css('.protractor-test-' + panelName + '-panel-insert-gadget-button'))
+    .click();
+
+  general.waitForSystem();
+
+  // Select the desired gadgetType from the modal.
+  element(
+    by.css('.protractor-test-' + gadgetType + '-gadget-selection-modal'))
+    .click();
+
+  var gadgetNameInput = element(by.model('gadgetDict.gadget_name'));
+  gadgetNameInput.clear().then(function() {
+    gadgetNameInput.sendKeys(gadgetName);
+  });
+
+  // Locate the customization section and apply any customizations.
+  var elem = element(by.css('.protractor-test-gadget-customization-editor'));
+  var customizationArgs = [elem];
+
+  if (arguments.length > 3) {
+    for (var i = 3; i < arguments.length; i++) {
+      customizationArgs.push(arguments[i]);
+    }
+    gadgets.getGadget(gadgetType).customizeGadget.apply(
+      null, customizationArgs);
+  }
+
+  element(by.css('.protractor-test-save-gadget-button')).click();
+
+  // Wait for the customization modal to close.
+  general.waitForSystem();
+
+};
+
+// currentName must exist.
+var renameGadget = function(currentName, newName) {
+  element(by.css('.protractor-test-rename-' + currentName + '-gadget-icon'))
+    .click();
+  var gadgetNameInput = element(by.model('newNameForRenamedGadget'));
+  gadgetNameInput.clear().then(function() {
+    gadgetNameInput.sendKeys(newName);
+  });
+  element(
+    by.css('.protractor-test-gadget-rename-confirmation-button')).click();
+};
+
+// gadgetName must exist.
+var deleteGadget = function(gadgetName) {
+  element(by.css('.protractor-test-delete-' + gadgetName + '-gadget-icon'))
+    .click();
+  general.waitForSystem(); // wait for modal popup.
+  element(by.cssContainingText('.btn-danger', 'Delete Gadget')).click();
```

Comment:
Again I would use a protractor-specific class and by.css here, as the user-facing text may change. See
https://github.com/oppia/oppia/wiki/Writing-Integration-Tests
for more guidelines on this.